deploy/tests/test_utils.py

```diff
@@ -0,0 +1,64 @@
+from pathlib import Path
+from unittest.mock import MagicMock, call
+from botocore.exceptions import ClientError
+import pytest
+import os
+
+from moto import mock_s3
+import boto3
+
+pytest.mark.usefixtures("aws_test_credentials")
+
+
+@pytest.mark.parametrize(
+    'protocol_type,test_dest_file_path',
+    [
+        ('s3',Path("tests/s3_test_download_json.json")),
+        ('file',Path("tests/file_test_download_json.json")),
```

Comment:
Do we want to add a test for ssh?  The simplest way to do it would be to spin up a docker container that the test would try to pull things out of with ssh.  If docker isn't installed, the test would be skipped.

Alternately, we write a test for this that utilizes the multiple hosts in CI.  Or we could do both so that devs can test this locally (if they have docker installed) and we have a test in CI without having to install docker in the place where we're running pytests.

Reply:
Couldn't we just test this by running `localhost` for the "it should work" case and some invalid IP for the "it shouldn't work" case? I'm not sure it's worth having docker installed for such a small test

Reply:
@abejgonzalez how do we differentiate whether someone doesn't have sshd configured correctly on localhost vs the test actually working?  I don't like having a dependency into sshd that the tests aren't actually controling the configuration for.

A lighter-weight option could be for a test fixture to spin up a Paramiko sshd on a non-system port or use something like [this gist](https://gist.github.com/vxgmichel/d0b7e4cc3caab32601051ee262ee7b31).   What do you think?

Reply:
That seems better to me.

Reply:
I've asked @filipstamenkovic-sifive to help me add a test.

Reply:
Captured in #1167  for followup.  I'd like to go ahead and get this merged if the CI passes this time.